client/main.go

```diff
@@ -1,232 +1,525 @@
-	return getCLI(networkName)
+func getConfigStore(dataDir string) (store.ConfigStore, error) {
+	return filestore.NewConfigStore(dataDir)
 }
 
-func getCLI(networkName string) (interfaces.CLI, error) {
-	switch networkName {
-	case common.Liquid.Name, common.LiquidTestNet.Name, common.LiquidRegTest.Name:
-		return covenant.New(), nil
-	case common.Bitcoin.Name, common.BitcoinTestNet.Name, common.BitcoinRegTest.Name, common.BitcoinSigNet.Name:
-		return covenantless.New(), nil
-	default:
-		return nil, fmt.Errorf("unknown network (%s)", networkName)
+func getNetwork(ctx *cli.Context, configData *store.StoreData) string {
+	if configData == nil {
+		return strings.ToLower(ctx.String("network"))
 	}
+	return configData.Network.Name
 }
 
-// cleanAndExpandPath expands environment variables and leading ~ in the
-// passed path, cleans the result, and returns it.
-// This function is taken from https://github.com/btcsuite/btcd
-func cleanAndExpandPath(path string) string {
-	if path == "" {
-		return ""
+func isBtcChain(network string) bool {
+	return network == common.Bitcoin.Name ||
+		network == common.BitcoinTestNet.Name ||
+		network == common.BitcoinRegTest.Name
+}
+
+func isLiquidChain(network string) bool {
+	return network == common.Liquid.Name ||
+		network == common.LiquidTestNet.Name ||
+		network == common.LiquidRegTest.Name
+}
+
+func parseReceivers(rcvrs string, isBitcoin bool) ([]arksdk.Receiver, error) {
+	if isBitcoin {
+		return arksdk.NewBitcoinReceiversFromJSON(rcvrs)
 	}
+	return arksdk.NewLiquidReceiversFromJSON(rcvrs)
+}
+
+func sendCovenantLess(ctx *cli.Context, receivers []arksdk.Receiver) error {
+	txID, err := arkSdkClient.SendAsync(
+		context.Background(),
+		ctx.Bool(enableExpiryCoinselectFlag.Name),
+		receivers,
+	)
+	if err != nil {
+		return err
+	}
+	return printJSON(map[string]interface{}{"txid": txID})
+}
 
-	// Expand initial ~ to OS specific home directory.
-	if strings.HasPrefix(path, "~") {
-		var homeDir string
-		u, err := user.Current()
-		if err == nil {
-			homeDir = u.HomeDir
+func sendCovenant(receivers []arksdk.Receiver) error {
+	var onchainReceivers, offchainReceivers []arksdk.Receiver
+
+	for _, receiver := range receivers {
+		if receiver.IsOnchain() {
+			onchainReceivers = append(onchainReceivers, receiver)
 		} else {
-			homeDir = os.Getenv("HOME")
+			offchainReceivers = append(offchainReceivers, receiver)
+		}
+	}
+
+	if len(onchainReceivers) > 0 {
+		txID, err := arkSdkClient.SendOnChain(context.Background(), onchainReceivers)
+		if err != nil {
+			return err
+		}
+		return printJSON(map[string]interface{}{"txid": txID})
+	}
+
+	if len(offchainReceivers) > 0 {
+		txID, err := arkSdkClient.SendOffChain(context.Background(), false, offchainReceivers)
+		if err != nil {
+			return err
 		}
+		return printJSON(map[string]interface{}{"txid": txID})
+	}
+
+	return nil
+}
 
-		path = strings.Replace(path, "~", homeDir, 1)
+func decrypt(encrypted, password []byte) ([]byte, error) {
```

Comment:
`decrypt` and `encrypt` are also implemented in the SDK package. Also they are used only to print the private key. Should we consider exporting the function or maybe implement `dumpPrivateKey` in sdk ? cc @altafan @sekulicd